sound/soc/sof/intel/hda-dsp.c

```diff
@@ -405,6 +405,26 @@ static int hda_dsp_update_d0i3c_register(struct snd_sof_dev *sdev, u8 value)
 	return 0;
 }
 
+static bool hda_dsp_d0i3_streaming(struct snd_sof_dev *sdev)
```

Comment:
we have snd_sof_dsp_only_d0i3_compatible_stream_active() for this already

Reply:
@ranj063  this function is used to detect playback stream with d0i3 enabled. For capture case we can't set streaming flag. do you have any idea to check with snd_sof_dsp_only_d0i3_compatible_stream_active ?

Reply:
why does the dfirection matter at all? if a stream is d0i3 compatible, that should be good enough no?

Reply:
@ranj063 like Pierre's comments "SRAM windows reporting link position have to remained powered-up".  SRAM is not powered up if only capture. If both are active we will have more hardware unit powered up